arch/xtensa/boot/dts/feathers3.dts

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-2.0
+/dts-v1/;
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/irq.h>
+#include "esp32s3.dtsi"
+
+/ {
+	memory@0 {
+		device_type = "memory";
+		reg = <0x3d800000 0x00800000>;
+	};
+
+	i2c0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "i2c-gpio";
+
+		sda-gpios = <&gpio0 8 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&gpio0 9 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+
+		pinctrl-0 = <&i2c0_pins>;
+		pinctrl-names = "default";
+    };
+
+    i2c1 {
+        #address-cells = <1>;
+        #size-cells = <0>;
+        compatible = "i2c-gpio";
+
+        sda-gpios = <&gpio0 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+        scl-gpios = <&gpio0 15 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+
+        pinctrl-0 = <&i2c1_pins>;
+        pinctrl-names = "default";
+    };
+};
+
+&chosen {
+    bootargs = "earlycon=esp32s3acm,mmio32,0x60038000 console=ttyACM1 debug rw root=mtd:rootfs no_hash_pointers";
+};
+
+&iomux {
+	i2c0_pins: i2c0_pins {
+		pinctrl-single,pins = <
+			PIN(8) FUN0_20MA_IE_WPU			/* SDA */
+			PIN(9) FUN0_20MA_IE_WPU>;		/* SCL */
+	};
+    i2c1_pins: i2c1_pins {
+		pinctrl-single,pins = <
+			PIN(16) FUN0_20MA_IE_WPU			/* SDA */
+			PIN(15) FUN0_20MA_IE_WPU>;		/* SCL */
+	};
+    spi2_pins: spi2_pins {
+		pinctrl-single,pins = <
+			PIN(5) (FUN_SEL(4) | FUN_DRV_20MA)	/* CS0 */
+			PIN(35) (FUN_SEL(4) | FUN_DRV_40MA)	/* MOSI */
+			PIN(36) (FUN_SEL(4) | FUN_DRV_40MA)	/* SCK */
+			PIN(37) (FUN_SEL(4) | FUN0_20MA_IE_WPU)>;/* MISO */
```

Comment:
Please see esp32s3 TRM table 6-3 for IOMUX functions. There's nothing on pin 5 function 4 and pins 35-37 function 4 are SPIIO6, SPIIO7 and SPIDQS, not SPI2 signals.